tensorboard/webapp/runs/store/runs_selectors.ts

```diff
@@ -72,27 +74,48 @@ export const getRun = createSelector(
  */
 export const getRuns = createSelector(
   getDataState,
-  (state: RunsDataState, props: {experimentId: string}): Run[] => {
+  getRunsToHparamsAndMetrics,
+  (
+    state: RunsDataState,
+    runsToHparamsAndMetrics: RunToHparamsAndMetrics,
+    props: {experimentId: string}
+  ): Run[] => {
     const runIds = state.runIds[props.experimentId] || [];
     return runIds
       .filter((id) => Boolean(state.runMetadata[id]))
-      .map((id) => state.runMetadata[id]);
+      .map((id) => ({...state.runMetadata[id]}))
+      .map((metadata) => {
+        if (runsToHparamsAndMetrics[metadata.id]) {
+          metadata.hparams = runsToHparamsAndMetrics[metadata.id].hparams;
+          metadata.metrics = runsToHparamsAndMetrics[metadata.id].metrics;
+        }
+        return metadata;
+      });
   }
 );
 
 export const getRunsFromExperimentIds = (experimentIds: string[]) =>
   createSelector(
     getDataState,
-    (state: RunsDataState): Array<Run & {experimentId: string}> => {
-      return experimentIds.reduce((runs, experimentId) => {
-        (state.runIds[experimentId] || [])
-          .filter((id) => Boolean(state.runMetadata[id]))
-          .forEach((runId) => {
-            runs.push({...state.runMetadata[runId], experimentId});
-          });
-
-        return runs;
-      }, [] as Array<Run & {experimentId: string}>);
+    getRunsToHparamsAndMetrics,
+    (
+      state: RunsDataState,
+      runsToHparamsAndMetrics: RunToHparamsAndMetrics
+    ): Array<Run & {experimentId: string}> => {
+      return experimentIds
```

Comment:
I think the restructured version of this selector is easier to read. I strongly considered chaining these two selectors together but opted not to as I hope to eventually replace `getRuns` with `getRunsFromExperimentIds` as we move away from using selectors with props.